geospaas_processing/cli/download.py

```diff
@@ -0,0 +1,87 @@
+"""
+Script for cli of downloading repo. This script accepts the argunemt from sys.args by the
+help of python "argparse" and download the dataset files based on the input arguments.
+"""
+import argparse
+import os
+import subprocess
+import sys
+import time
+from datetime import datetime
+from distutils.util import strtobool
+import json
+import django
+import ast
+
+from dateutil.relativedelta import relativedelta
+from dateutil.tz import tzutc
+from django.contrib.gis.geos import GEOSGeometry
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_processing.settings')
+django.setup()
+from geospaas.catalog.models import Dataset
+import geospaas_processing.downloaders as downloaders
+
+
+def main(ar):
+    cumulative_limitation = {}
+    limitation_list = json.loads(ar.limitation)
+    for limitation in limitation_list:
+        cumulative_limitation.update(limitation)
+    if ar.rel_time_flag:
+        designated_begin = datetime.now().replace(tzinfo=tzutc()) + relativedelta(
+            hours=-abs(int(ar.begin)))
+        designated_end = datetime.now().replace(tzinfo=tzutc())
+    else:
+        designated_begin = datetime.strptime(ar.begin, "%Y.%m.%d:%H").replace(
+            tzinfo=tzutc())
+        designated_end = datetime.strptime(ar.end, "%Y.%m.%d:%H").replace(tzinfo=tzutc())
+    download_manager = downloaders.DownloadManager(
+        download_directory=ar.down_dir.rstrip(os.path.sep)+os.path.sep+ar.dir_struct.lstrip(
+            os.path.sep),
+        provider_settings_path=ar.config_file,
+        # 400 dataset per day is allowed
+        max_downloads=400*(abs((designated_end-designated_begin).days)+1),
+        use_file_prefix=ar.use_filename_prefix,
+        time_coverage_start__gte=designated_begin,
+        time_coverage_end__lte=designated_end,
+        geographic_location__geometry__intersects=GEOSGeometry(ar.geometry),
+        **cumulative_limitation
+    )
+    download_manager.download()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
```

Comment:
Create a separate function for parsing arguments. Then it will be easier for testing.

Reply:
Done in the next commit